src/Shared/AssemblyUtilities.cs

```diff
@@ -39,5 +39,23 @@ public static Type GetTypeInfo(this Type t)
             return t;
         }
 #endif
+
+        public static AssemblyName Clone2(this AssemblyName assemblyNameToClone)
+        {
+#if FEATURE_ASSEMBLYNAME_CLONE
+            return assemblyNameToClone.Clone();
+#else
+            AssemblyName result = null;
+            try
+            {
+                MethodInfo method = assemblyNameToClone?.GetType().GetMethod("Clone");
```

Comment:
Not caching reflection results is quite slow too. You can use static field to store `typeof(AssemblyName).GetType().GetMethod("Clone")` result.